pallets/parachain-staking/src/lib.rs

```diff
@@ -1828,16 +1830,25 @@ pub mod pallet {
 						let percent = Perbill::from_rational(amount, state.total);
 						let due = percent * amt_due;
 						if !due.is_zero() {
-							extra_weight = extra_weight.saturating_add(Self::mint_and_compound(
```

Comment:
We should change `mint_and_compound` to not return `Weight` if it's unused, could be done as a follow-up